### PR TITLE
improvement: Show direct members first

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would show all members irrelevant if they were actually declared in the class or parents such as Object. Now, we show direct members first.

Also, changed the compielr to not report errors while typing.

### CODE REVIEW

File: mtags-java/src/main/scala/scala/meta/internal/pc/JavaCompletionProvider.scala [hunk 1]
```diff
@@ -52,32 +53,27 @@ class JavaCompletionProvider(
...
-
-      override def compare(i1: CompletionItem, i2: CompletionItem): Int = {
-        java.lang.Integer.compare(score(i1), score(i2))
-      }
-    }
+  private def identifierScore(element: Element): Int = {
+    val name = element.getSimpleName().toString().toLowerCase()
+    if (name.startsWith(identifier)) 0
+    else if (name.contains(identifier)) 1
+    else 2
```
Comment (dos65) [inline mtags-java/src/main/scala/scala/meta/internal/pc/JavaCompletionProvider.scala:69]: Optional optimization
```suggestion
   name.indexOf(identifier) match {
     case 0 => 0
     case -1 => 2
     case _ => 1
   }
```

**Review by dos65 — APPROVED** [review]
:+1:

File: mtags-java/src/main/scala/scala/meta/internal/pc/JavaMetalsGlobal.scala [hunk 2]
```diff
@@ -19,14 +22,21 @@ class JavaMetalsGlobal(
...
+
   def compilationTask(sourceCode: String, uri: URI): JavacTask = {
     val javaFileObject = SourceJavaFileObject.make(sourceCode, uri)
 
     COMPILER
       .getTask(
         null,
         null,
-        null,
+        noopDiagnosticListener,
```
Comment (kasiaMarek) [inline mtags-java/src/main/scala/scala/meta/internal/pc/JavaMetalsGlobal.scala:39]: Did it report the errors somewhere when it was a null?

**Review by kasiaMarek — APPROVED** [review]
lgtm

File: mtags-java/src/main/scala/scala/meta/internal/pc/JavaMetalsGlobal.scala (hunk 2 above)
Comment (tgodzik) [reply mtags-java/src/main/scala/scala/meta/internal/pc/JavaMetalsGlobal.scala:39]: It reported it to stderr, co would end up in Metals output, which is not really useful

**Pull request merged**